### PR TITLE
feat(http): map uploaded files into the request properties

### LINKED ISSUE #370: Map Upload files into request properties
```php
class MyRequest implements Request
{
    use IsRequest;
    
    public Upload $avatar;
}
```

### PR DESCRIPTION
**Pull request by yassiNebeL**
fixed #370

### CODE REVIEW

File: src/Tempest/Mapper/src/Mappers/ArrayToObjectMapper.php [hunk 1]
```diff
@@ -160,9 +160,9 @@ private function resolveValueFromArray(
 
         $caster = $this->casterFactory->forProperty($property);
 
-        foreach ($data as $item) {
+        foreach ($data as $key => $item) {
             if (! is_array($item)) {
-                $values[] = $caster?->cast($item) ?? $item;
+                $values[$key] = $caster?->cast($item) ?? $item;
```
Comment (brendt) [inline src/Tempest/Mapper/src/Mappers/ArrayToObjectMapper.php:165]: Why was this change necessary?

File: src/Tempest/Http/src/Mappers/PsrRequestToRequestMapper.php [hunk 2]
```diff
@@ -53,7 +53,7 @@ public function map(mixed $from, mixed $to): array|object
             'path' => $from->getUri()->getPath(),
             'query' => $query,
             'files' => $uploads,
-            ...$data,
+            ...$data + $uploads,
```
Comment (brendt) [inline src/Tempest/Http/src/Mappers/PsrRequestToRequestMapper.php:56]: Could you write a test?

Comment (brendt) [reply src/Tempest/Http/src/Mappers/PsrRequestToRequestMapper.php:56]: Also, I'm personally more a fan on `[...$data, ...$uploads]`, since `[] + []` works a bit weird in PHP

**Review by brendt — CHANGES_REQUESTED** [review]

File: src/Tempest/Mapper/src/Mappers/ArrayToObjectMapper.php (hunk 1 above)
Comment (yassiNebeL) [reply src/Tempest/Mapper/src/Mappers/ArrayToObjectMapper.php:165]: this is necessary when you have a input file with a name, so this ensures that the array is associative, please check the related issue.

File: src/Tempest/Http/src/Mappers/PsrRequestToRequestMapper.php (hunk 2 above)
Comment (yassiNebeL) [reply src/Tempest/Http/src/Mappers/PsrRequestToRequestMapper.php:56]: OK. can I use the `FormRequest` for testing?

**Comment by yassiNebeL** [comment]
I updated the PR regarding the changes, however, the merging is still blocked.

**Comment by brendt** [comment]
Could you please take a look at the failed CI runs?

**Comment by yassiNebeL** [comment]
could you please review the PR now? all CI checks have passed!
